### ISSUE TITLE
Show currently processed file in -x

### ISSUE THREAD

**Issue by hyperknot** [sharkdp/fd#963]
I'm running a large batch processing job using -x. It'd be nice to display the name of the currently processed file or files (with parallel processing).

I don't know any other way to achieve this. If there is a workaround please let me not.

**Comment by tmccombs** [comment]
Currently, the only way to do this is with something like `bash -c 'echo {}; my-command {}'`. However, once https://github.com/sharkdp/fd/pull/960 is merged you could do it with `-x echo {} \; -x my-command...`. 

I also think this is a duplicate of #943

**Comment by hyperknot** [comment]
I understand. Just a small proposal for `--print-filename` or similar which would print the filename without any of the complexity described above.

**Comment by tmccombs** [comment]
is `-x echo \;` that much more complicated than `--print-filename`?

**Comment by hyperknot** [comment]
Yes, personally I'd find it more complicated. + `--print-filename` could use color output.

**Comment by sharkdp** [comment]
This option seems to be requested a lot, given the amount of related discussions:
- https://github.com/sharkdp/fd/issues/963
- https://github.com/sharkdp/fd/issues/959
- https://github.com/sharkdp/fd/issues/943
- https://github.com/sharkdp/fd/pull/946
- https://github.com/sharkdp/fd/issues/406

@tmccombs Multi-exec is a fantastic feature which I'm very excited about, but maybe we should reconsider adding a simple option for this? Proper formatting would be one argument. Accessibility of the feature another.

**Comment by tmccombs** [comment]
Yeah, I'm persuaded by the formatting/color option.

**Comment by vladh** [comment]
I would like to second the request for this feature, with a slightly more detailed rationale.

When using the `-x` option, one is often dealing with a lot of files, and often performing destructive operations on those files.

In this scenario, I really do not want to add any other `echo` calls or any superfluous syntax to the command I am passing to `-x`, since I do not want to risk getting something wrong and losing data.

Having a simple flag to enable printing would make this common use-case much easier, without presenting the risk of the user accidentally getting the command wrong in the process.

I would even go as far as to say that I think this should be the default behaviour — it's important to know what commands are being run on your files.

**Comment by sharkdp** [comment]
I think we can close this in favor of #1042. Let me know if anyone disagrees.

**Comment by Kristinita** [comment]
**Type: Reply** 💬

### 1. Summary

> is `-x echo \;` that much more complicated than `--print-filename`?

@tmccombs, `-x echo \;` isn’t cross-platform syntax. `\;` works on UNIX, `;` on Windows.

### 2. MCVE

#### 2.1. Online demonstration

My MCVE on CI services:

1. [**Travis CI build**](https://app.travis-ci.com/github/Kristinita/SashaTravis/builds/269234128) on Ubuntu 22.04.3 LTS Jammy Jellyfish
1. [**AppVeyor CI build**](https://ci.appveyor.com/project/Kristinita/sashatravis/builds/49313901) on Windows Server 2019

#### 2.2. Files

`KiraValidFile.sh`:

```sh
#!/bin/bash

```

`KiraNonValidFile.sh`:

```sh
#!/bin/Kira Goddess!

```

#### 2.3. Commands

```sh
# First command
- fd --extension sh --exec echo ; --exec shellcheck

# Second command
- fd --extension sh --exec echo \; --exec shellcheck
```

#### 2.4. Behavior

##### 2.4.1. First command

1. [**Ubuntu**](https://app.travis-ci.com/github/Kristinita/SashaTravis/builds/269234128#L202):

	```sh
	$ fd --extension sh --exec echo ; --exec shellcheck

	./KiraValidFile.sh

	./KiraNonValidFile.sh

	--exec: command not found

	The command "fd --extension sh --exec echo ; --exec shellcheck" exited with 127.
	```

1. [**Windows**](https://ci.appveyor.com/project/Kristinita/sashatravis/builds/49313901#L43):

	```sh
	fd --extension sh --exec echo ; --exec shellcheck
	.\KiraValidFile.sh
	.\KiraNonValidFile.sh
	In .\KiraNonValidFile.sh line 1:
	#!/bin/Kira Goddess!
	^-- SC1008 (error): This shebang was unrecognized. ShellCheck only supports sh/bash/dash/ksh. Add a 'shell' directive to specify.
	For more information:
	  https://www.shellcheck.net/wiki/SC1008 -- This shebang was unrecognized. Sh…
	Command exited with code 
	```

First command works for Windows, but doesn’t work for Ubuntu.

##### 2.4.2. Second command

1. [**Ubuntu**](https://app.travis-ci.com/github/Kristinita/SashaTravis/builds/269234128#L207):

	```sh
	$ fd --extension sh --exec echo \; --exec shellcheck

	./KiraValidFile.sh

	./KiraNonValidFile.sh

	In ./KiraNonValidFile.sh line 1:

	#!/bin/Kira Goddess!

	^-- SC1008: This shebang was unrecognized. ShellCheck only supports sh/bash/dash/ksh. Add a 'shell' directive to specify.

	For more information:

	  https://www.shellcheck.net/wiki/SC1008 -- This shebang was unrecognized. Sh…

	The command "fd --extension sh --exec echo \; --exec shellcheck" exited with 1.
	```

1. [**Windows**](https://ci.appveyor.com/project/Kristinita/sashatravis/builds/49313901#L54):

	```sh
	fd --extension sh --exec echo \; --exec shellcheck
	\; --exec shellcheck .\KiraValidFile.sh
	\; --exec shellcheck .\KiraNonValidFile.sh
	```

Second command works for Ubuntu, but doesn’t work for Windows.

### 3. About cross-platform syntax

When I read this issue tracker, it seems to me that in most discussions here users and developers discussing the behavior of fd in UNIX shells without considering the Windows command prompt syntax. Users who want cross-platform syntax support may be in the minority here, but we exist. I would like fd to have more cross-platform syntax so that users don’t have to write different commands for different operating systems. Especially for such frequently needed tasks as printing filename.

Thanks.

**Comment by Kristinita** [comment]
**Type: Reply** 💬

> I think we can close this in favor of #1042. Let me know if anyone disagrees.

@sharkdp, **#1042** and **#1068** about printing the executed commands, this issue about printing the filenames. I would prefer fd to print filenames rather than the same executed command for all files. If fd had a CLI argument for printing filenames, maybe users won’t be upset.

Thanks.
